server/src/main/java/org/elasticsearch/common/time/DateUtils.java

```diff
@@ -129,6 +129,27 @@ public static Instant toInstant(long nanoSecondsSinceEpoch) {
         return Instant.ofEpochSecond(seconds, nanos);
     }
 
+    /**
+     * Convert a nanosecond timestamp in milliseconds
+     *
+     * @param milliSecondsSinceEpoch the millisecond since the epoch
+     * @return                      the nanoseconds since the epoch
+     */
+    public static long toNanoSeconds(long milliSecondsSinceEpoch) {
+        if (milliSecondsSinceEpoch < 0) {
+            throw new IllegalArgumentException("milliSeconds [" + milliSecondsSinceEpoch + "] are before the epoch in 1970 and cannot " +
+                "be converted to nanoseconds");
+        } else if (milliSecondsSinceEpoch > MAX_NANOSECOND_INSTANT.toEpochMilli()) {
+            throw new IllegalArgumentException("milliSeconds [" + milliSecondsSinceEpoch + "] are after 2262-04-11T23:47:16.854775807 " +
+                "and cannot be converted to nanoseconds");
+        }
+
+        if (milliSecondsSinceEpoch == 0) {
+            return 0;
+        }
```

Comment:
why this special case?

Reply:
just a copy/paste from the other conversion, this is not needed